File: LLM-Translations/GPT-4/codenet/C++/C/s847126533.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>

bool check(int n, int wei, int* book, int book_size)
{
  int tmp=wei;
  n--;
  for(int i=0;i<book_size;i++)
    {
      if(tmp<book[i])
	{
	  if(n>0)
	    {
	      n--;
	      tmp=wei;
	      if(tmp<book[i])
		return false;
	    }
	  else
	    return false;
	}
      tmp-=book[i];
    }
  return true;
}

int main()
{
  int m,n;
  while(scanf("%d %d", &m, &n),m)
    {
      int* book = (int*) malloc(n * sizeof(int));
      for(int i=0;i<n;i++)
	{
	  int tmp;
	  scanf("%d", &tmp);
	  book[i] = tmp;
	}

      int l=0;
      int r=1500000;
      int ans;
      while(l<r)
	{
	  if(check(m,(l+r)/2,book,n))
	    {
	      ans=(l+r)/2;
	      if(r==(l+r)/2)
		break;
	      r=(l+r)/2;

	    }
	  else
	    {
	      if(l==(l+r)/2)
		break;
	      l=(l+r)/2;
	    }
	}
      printf("%d\n", ans);
      free(book);
    }
    return 0;
}
// 

File: LLM-Translations/GPT-4/codenet/C++/Python/s847126533.py

def check(n, wei, book):
    tmp = wei
    n -= 1
    for i in range(len(book)):
        if tmp < book[i]:
            if n > 0:
                n -= 1
                tmp = wei
                if tmp < book[i]:
                    return False
            else:
                return False
        tmp -= book[i]
    return True

def main():
    while True:
        m, n = map(int, input().split())
        if m == 0:
            break
        book = [int(input()) for _ in range(n)]

        l = 0
        r = 1500000
        ans = 0

        while l < r:
            mid = (l + r) // 2
            if check(m, mid, book):
                ans = mid
                if r == mid:
                    break
                r = mid
            else:
                if l == mid:
                    break
                l = mid

        print(ans)

if __name__ == "__main__":
    main()


